flaskr/templates/blog/index.html

```diff
@@ -18,11 +18,24 @@ <h1>{{ post['title'] }}</h1>
         {% if g.user['id'] == post['author_id'] %}
           <a class="action" href="{{ url_for('blog.update', id=post['id']) }}">Edit</a>
         {% endif %}
+        <button class="toggle-content" onclick="toggleContent(this)">Expand</button>
       </header>
-      <p class="body">{{ post['body'] }}</p>
+      <p class="body" style="display: none;">{{ post['body'] }}</p>
     </article>
     {% if not loop.last %}
       <hr>
     {% endif %}
   {% endfor %}
+  <script>
+    function toggleContent(button) {
```

Comment:
**Issue**: Avoid directly manipulating the `style` attribute in JavaScript. Instead, toggle a CSS class that controls the visibility of the post content. This approach will make it easier to manage styles and maintain the code.

**Fix**: function toggleContent(button) {
  const body = button.parentElement.nextElementSibling;
  body.classList.toggle('hidden');
  button.textContent = body.classList.contains('hidden') ? 'Expand' : 'Collapse';
}